core/iwasm/fast-jit/fe/jit_emit_numberic.c

```diff
@@ -173,9 +209,25 @@ compile_int_mul(JitCompContext *cc, JitReg left, JitReg right, bool is_i32)
     if (IS_CONST_ZERO(left) || IS_CONST_ZERO(right))
         return is_i32 ? NEW_CONST(I32, 0) : NEW_CONST(I64, 0);
 
+    if (jit_reg_is_const(left) && jit_reg_is_const(right)) {
+        if (is_i32) {
+            int32 left_val = jit_cc_get_const_I32(cc, left);
+            int32 right_val = jit_cc_get_const_I32(cc, right);
+            res = NEW_CONST(I32, left_val * right_val);
+        }
+        else {
+            int64 left_val = jit_cc_get_const_I64(cc, left);
+            int64 right_val = jit_cc_get_const_I64(cc, right);
+            res = NEW_CONST(I64, left_val * right_val);
+        }
+        goto shortcut;
+    }
```

Comment:
How about define these code piece to a macro, e.g. `CHECK_CONST(left, right, op)`?

Reply:
Yes, how about adding a function to do that:
```C
bool check_const_and_operate(JitReg left, JitReg right, ArithOp op, JitReg *p_res)
{
    if (jit_reg_is_const(left) && jit_reg_is_const(right)) {
        if (is_i32) {
            int32 left_val = jit_cc_get_const_I32(cc, left);
            int32 right_val = jit_cc_get_const_I32(cc, right);
            int32 res_val;
            switch (op) {
                 case ADD: res_val = left_val + right_val;
                  break;
            res = NEW_CONST(I32, res_val);
        }
        else {
            int64 left_val = jit_cc_get_const_I64(cc, left);
            int64 right_val = jit_cc_get_const_I64(cc, right);
            ...
        }
        *p_res = res;
        return true;
    }
    return false;
}

Reply:
Sure. working on it.